src/main/java/ca/on/oicr/pineryreports/reports/impl/OctaneCountsReport.java

```diff
@@ -0,0 +1,487 @@
+      new ColumnDefinition("")
+  ));
+  
+  private String start;
+  private String end;
+  
+  private final List<Count> caseNumbers = new ArrayList<>();
+  private final List<Count> sampleNumbers = new ArrayList<>();
+  private List<Count> inventoryNumbers = new ArrayList<>();
+  private List<Count> tissueTypeNumbers = new ArrayList<>();
+  private List<Count> tissueOriginNumbers = new ArrayList<>();
+  
+  @Override
+  public String getReportName() {
+    return REPORT_NAME;
+  }
+
+  @Override
+  public Collection<Option> getOptions() {
+    Set<Option> opts = Sets.newHashSet();
+    
+    opts.add(Option.builder()
+        .longOpt(OPT_AFTER)
+        .hasArg()
+        .argName("date")
+        .desc("Report receipt/creation after this date, inclusive (yyyy-mm-dd)")
+        .build());
+    opts.add(Option.builder()
+        .longOpt(OPT_BEFORE)
+        .hasArg()
+        .argName("date")
+        .desc("Report receipt/creation before this date, exclusive (yyyy-mm-dd)")
+        .build());
+    
+    return opts;
+  }
+
+  @Override
+  public void processOptions(CommandLine cmd) throws ParseException {
+    if (cmd.hasOption(OPT_AFTER)) {
+      String after = cmd.getOptionValue(OPT_AFTER);
+      if (!after.matches(DATE_REGEX)) {
+        throw new ParseException("After date must be in format yyyy-mm-dd");
+      }
+      this.start = after;
+    }
+    
+    if (cmd.hasOption(OPT_BEFORE)) {
+      String before = cmd.getOptionValue(OPT_BEFORE);
+      if (!before.matches(DATE_REGEX)) {
+        throw new ParseException("Before date must be in format yyyy-mm-dd");
+      }
+      this.end = before;
+    }
+  }
+
+  @Override
+  public String getTitle() {
+    return "OCTANE Counts "
+        + (start == null ? "Any Time" : start)
+        + " - "
+        + (end == null ? "Now" : end)
+        + ", and Inventory "
+        + new SimpleDateFormat("yyyy-MM-dd").format(new Date());
+  }
+  
+  private static final String ATTR_CATEGORY = "Sample Category";
+  private static final String ATTR_TISSUE_TYPE = "Tissue Type";
+  private static final String ATTR_TISSUE_ORIGIN = "Tissue Origin";
+  private static final String ATTR_SLIDES = "Slides";
+  private static final String ATTR_DISCARDS = "Discards";
+  private static final String ATTR_CONSUMED = "Slides Consumed";
+  private static final String ATTR_REMAINING = "Remaining";
+  private static final String ATTR_STAIN = "Stain";
+  
+  private static final String SAMPLE_CLASS_SLIDE = "Slide";
+  private static final String SAMPLE_CLASS_WHOLE_RNA = "whole RNA";
+  
+  private static final String SAMPLE_CATEGORY_IDENTITY = "Identity";
+  private static final String SAMPLE_CATEGORY_TISSUE = "Tissue";
+  private static final String SAMPLE_CATEGORY_STOCK = "Stock";
+  private static final String SAMPLE_CATEGORY_ALIQUOT = "Aliquot";
+
+  private static final String STAIN_HE = "Hematoxylin+Eosin";
+  
+  @Override
+  protected void collectData(PineryClient pinery) throws HttpResponseException {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 39 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)